### PR TITLE
🏗 Set up codebase

### PR DESCRIPTION
**Pull request by hinosxz**
# Changes
Adds:
- pre-commit hooks with black code formatter
- requirements.txt file
- setup documentation
- pull request template

# Testing
Try pulling the branch and:
- running `pip install -r requirements.txt`
- committing an empty file, black should work

### CODE REVIEW

**Review by qchuchu — COMMENTED** [review]
RAS - Mais pas tout compris tous les fichiers xml que tu as rajouté

**Comment by hinosxz** [comment]
Oops, sorry about that. I should've explained in the PR description. I uploaded my `idea` folder that contains PyCharm config files. I thought they weren't worth putting in the `.gitignore`. Please tell me if you think otherwise 🙂

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# Changes
+<!-- Replace this line with a description of the changes you made -->
+
+# Testing
+<!-- Replace this line with a quick way of testing your changes work as expected -->
```
Comment (qchuchu) [inline .github/pull_request_template.md:5]: Donc ça c'est un document que tu mets à chaque PR ? Mais genre tu les stockes où ? Parce que si chacun fait une PR ça va faire des tonnes et des tonnes de fichier non ?

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+-   repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v2.3.0
+    hooks:
+    -   id: check-yaml
+    -   id: end-of-file-fixer
+    -   id: trailing-whitespace
+-   repo: https://github.com/psf/black
+    rev: 19.3b0
+    hooks:
+    -   id: black
```
Comment (qchuchu) [inline .pre-commit-config.yaml:11]: OK ! So this file is added to add a new hook that will be run each time someone try to commit. But how git knows that he has to read this file before commiting ?

Comment (qchuchu) [reply .pre-commit-config.yaml:11]: Aaaaah my bad I didn't read the pre-commit doc before commenting. But so we need to do `pre-commit install` on our project in order to install the hook ?

File: .github/pull_request_template.md (hunk 1 above)
Comment (hinosxz) [reply .github/pull_request_template.md:5]: No, this is only the pull request template. It'll show up as default content when you create a PR on the repo.

File: .pre-commit-config.yaml (hunk 2 above)
Comment (hinosxz) [reply .pre-commit-config.yaml:11]: Exactly 👍  I should add that to the readme!

**Review by qchuchu — APPROVED** [review]

**Pull request merged**